app/models/question_option.rb

```diff
@@ -49,6 +49,12 @@ class QuestionOption < ApplicationRecord
   validates :is_default, inclusion: { in: BOOLEAN_VALUES,
                                       message: INCLUSION_MESSAGE }
 
+  # =============
+  # = Callbacks =
+  # =============
+
+  # TODO: condition.option_list needs to be serialized (from Array) before we can check
+  # for related conditions, so this can't be replaced by :destroy on the association
   before_destroy :check_condition_options
```

Comment:
See the above comment on `app/models/question.rb` as this is effectively the same but on the `condition.option_list` attribute